FunctionalTests/Microsoft.Bot.Builder.FunctionalTests/FacebookChatTests.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.FunctionalTests
+{
+    [TestClass]
+    [TestCategory("FunctionalTests")]
+    public class FacebookChatTests
+    {
+        private string _appSecret;
+        private string _accessToken;
+        private string _botEndpoint;
+        private string _senderId;
+
+        [TestMethod]
+        public async Task SendAndReceiveFacebookMessageShouldSucceed()
+        {
+            GetEnvironmentVars();
+            string echoGuid = Guid.NewGuid().ToString();
+            await SendMessageAsync(echoGuid);
+            string response = await ReceiveMessageAsync();
+
+            Assert.AreEqual($"Echo: {echoGuid}", response);
+        }
+
+        private async Task SendMessageAsync(string echoGuid)
+        {
+            string bodyMessage = CreateBodyMessage(echoGuid);
+            string hubSignature = CreateHubSignature(bodyMessage);
+            var client = new HttpClient();
+            var request = new HttpRequestMessage();
+            request.Headers.Add("user-agent", "facebookexternalua");
+            request.Headers.Add("x-hub-signature", hubSignature);
+            request.Content = new StringContent(bodyMessage, Encoding.UTF8, "application/json");
+            request.Method = HttpMethod.Post;
+            request.RequestUri = new Uri(_botEndpoint);
+
+            await client.SendAsync(request);
+        }
+
+        private async Task<string> ReceiveMessageAsync()
+        {
+            await Task.Delay(TimeSpan.FromSeconds(10)).ConfigureAwait(false);
+            string retrieveMessgesUri = $"https://graph.facebook.com/v5.0/me/conversations?fields=messages{{message}}&user_id={_senderId}&access_token={_accessToken}";
+            var client = new HttpClient();
```

Comment:
Disposable objects should be initialized like this:
```suggestion
            using (var client = new HttpClient())
            {
            }
```